### ISSUE TITLE
Authorizes Count: Dynamic or Fixed

### ISSUE THREAD

**Issue by wizd** [graft-project/LyraNetwork#9]
The consensus algorithm used by Lyra Permissionless is pBFT, which needs to know the number of validator nodes. Slava defines the number of 21, which means 450+ messages to be transmitted per one block. it is a lot of bandwidth.

Another concern of Lyra is when something bad happens the count of nodes can't meet the predefined number, it makes the network not working anymore.

If Lyra uses a dynamic model, like

Win Number = (Total Qualified Nodes Count / 3) * 2 + 1

It will well tolerant of the changes of nodes count

**Comment by jpass022** [comment]
Dynamic looks to be a more robust model to use... I vote Dynamic.

**Comment by wizd** [comment]
> Dynamic looks to be a more robust model to use... I vote Dynamic.

Yes, dynamic mode looks more like a traditional blockchain. one bad thing it brings is alt chain reorg.

**Comment by RoggyDog** [comment]
Wouldn't dynamic have the potential to slow down consensus in the case that many nodes come online, and also increase bandwidth if there are like 100+ nodes? Don't we possibly run into a situation where the slow, under powered nodes make the network not as robust? Shouldn't the dynamic model still have a max set for nodes... If (Total Qualified Nodes Count < 21) {  Win Number = (Total Qualified Nodes Count / 3) * 2 + 1 } else { Win Number = 15 and choose 21 validator nodes from qualified nodes count }

**Comment by jpass022** [comment]
IMO network shouldn't allow under powered nodes.

**Comment by wizd** [comment]
Lyra nodes bootstrap procedure:

1) genesis with 4 seed nodes. the genesis wallet will automatically send funds to 10+ nodes' wallets.
2) authorizers count will keep increment until it reaches 21. (never turn back to smaller number)
3) when nodes count goes 21+, only select the first 21 qualified authorizers which ordered by staking amount. (yes, bellow 21 is now allowed)

Normal transaction begin with step 2.
